ipykernel/kernelbase.py

```diff
@@ -76,6 +76,8 @@ class Kernel(SingletonConfigurable):
     # attribute to override with a GUI
     eventloop = Any(None)
 
+    processes = {}
```

Comment:
Why is this map needed?

Reply:
Maybe not a big deal, but this seems a bit "append" only (i.e. a process object is never removed)

Reply:
> Maybe not a big deal, but this seems a bit "append" only (i.e. a process object is never removed)

https://github.com/ipython/ipykernel/pull/937/commits/17358fb26e2a362691c3f59e4aa0615d6abaf50c addresses this.